src/routing_core.rs

```diff
@@ -42,17 +48,20 @@ pub struct RoutingCore {
     network_name  : Option<NameType>,
     routing_table : Option<RoutingTable>,
     relay_map     : RelayMap,
+    // sender for signaling churn events
+    event_sender  : Sender<Event>,
 }
 
 impl RoutingCore {
     /// Start a RoutingCore with a new Id and the disabled RoutingTable
-    pub fn new() -> RoutingCore {
+    pub fn new(event_sender : Sender<Event>) -> RoutingCore {
         let id = Id::new();
         RoutingCore {
             id            : id,
             network_name  : None,
             routing_table : None,
             relay_map     : RelayMap::new(),
+            event_sender  : event_sender,
```

Comment:
I would have preferred if this struct stayed "just" a database of connections. Letting it know about the `event_sender` adds to responsibilies of this class.